tritium-metrics/src/main/metrics/metrics.yml

```diff
@@ -86,3 +86,16 @@ namespaces:
         type: histogram
         tags: [executor]
         docs: A histogram of the time it took to run a fixed-rate scheduled task as a percentage of the scheduled rate. Applies only to scheduled executors.
+      # ThreadFactory metrics
+      threads.created:
+        type: meter
+        tags: [executor]
+        docs: Rate that new threads are created for this executor.
+      threads.terminated:
+        type: meter
+        tags: [executor]
+        docs: Rate that executor threads are terminated.
+      threads.running:
+        type: counter
+        tags: [executor]
+        docs: Number of live threads created by this executor.
```

Comment:
ThreadFactory metrics are in the executor section, which isn't necessarily accurate, but it's where we tend to use them. For metric consumers I think this might be easier to follow than separate components and `thread-factory` vs `executor` tag names, but I don't have a strong opinion.

Looking for feedback on this choice.

Reply:
The tagging might be a bit confusing for direct consumers but for those that get instrumentation for free through witchcraft I think this choice will make thing simpler. I expect the later to be the common case so I'm ok with it as is.

Reply:
I'm going to hold off merging for a bit to give @schlosna a chance to weigh in on this.

Reply:
I'm fine with using `executor` name in this context for consistency as I'd expect all use of instrumented thread factory via WC or similar executor factory.